uaclient/entitlements/base.py

```diff
@@ -445,6 +454,13 @@ def enable(
         if not util.handle_message_operations(msg_ops):
             return False, None
 
+        # TODO: Move all logic from RepoEntitlement that
+        # handles the additionalPackages and APT directives
+        # to the base class. The additionalSnaps handling
+        # is a step on that direction
+        if not self.handle_additional_snaps():
+            return False, None
+
         ret = self._perform_enable(silent=silent)
         if not ret:
             return False, None
```

Comment:
That we're installing snaps before enabling is interesting!

`additionalPackages` are installed after enabling, so it's slightly weird that they are handled differently.

But canonical-livepatch is similar, it is a snap installed before enabling, so this fits with the livepatch service.

I think this is good then - just an observation. And maybe: should we change the name from `additionalSnaps` to `requiredSnaps` or otherwise to make it clear that it is handled differently? (I think that may have been the original name :upside_down_face:)

Reply:
Yes, as we have discussed, let's change this to use `requiredSnaps` instead